### DISCUSSION TITLE
Why rejection angles for wheels sweep hits are missing in vehicle2?

### DISCUSSION THREAD

**Discussion by Tooflevor** [NVIDIA-Omniverse/PhysX#77] [Q&A] [answered]
There was a cool feature for wheels sweeps in PhysX4 described [here](https://gameworksdocs.nvidia.com/PhysX/4.1/documentation/physxguide/Manual/Vehicles.html#wheel-contact-beyond-raycasts). It's pretty simple, but really powerful.
I didn't find anything like this in vehicle2 from PhysX5.
So why is it missing? Are there any downfalls of this feature?

**Comment by msauter-nvidia** [comment, marked as answer]
There are two main reasons for not offering this in PhysX 5 any longer. One is that the contact modification callback is user-owned and the way the data is organized can be very application specific, in addition, the rejecting logic also might need customization depending on the specific application. Since the vehicle SDK in 5.0 has a much more modular API, it is now possible to more easily create a custom component that implements a user defined rejection logic.
The second reason is that there are plans to investigate an option of using articulations or similar to model the wheel and suspension system for setups where the wheels are not protected by a chassis. This might then re-introduce something that is conceptually similar to the rejection angle approach.

**Reply by Tooflevor** [reply]
Am I understand it right: To implement this logic now, I need to create my own component `PxVehiclePhysXRoadGeometrySceneQueryComponent` with my own `PxVehiclePhysXRoadGeometryQueryUpdate` function?  
Because I don't see other way to reject some sweep hits. The best way would be to use filter, but I see no way to define which wheel is beeing cast in filter.

**Reply by msauter-nvidia** [reply]
That's right, you would implement your own component similar to PxVehiclePhysXRoadGeometrySceneQueryComponent and add the rejection logic you want.

When you say: the best way would be to use filter data/callback. Assuming we had filter data per wheel, what would be your plan? Store the wheel info in the filter data and the rejection angles in the callback and then use postFilter() for your rejection logic? Sounds like a good enough use case for a discussion whether adding per wheel filter data is something to consider in future versions. With the modular design it is possible to add the logic already but I can see that it would be more convenient (less work) through filter data.
I guess you could already work with the filter callback now, it would just be a bit hacky since you would need to figure out the wheel based on the hit point location.

**Reply by Tooflevor** [reply]
It's like you've read my mind😀 It's exactly what I'd like. I ended up with this exact hack: I am identifying the wheel by looking for the closest one to a hit point in postfilter. And yes, filter per wheel would solve my issue.

**Reply by msauter-nvidia** [reply]
OK. Thanks for this feedback. We will discuss it for sure. Can't promise more than that though.

**Reply by msauter-nvidia** [reply]
update: per wheel filter data will be supported in PhysX 5.2.0
